### ISSUE TITLE
OrchestrationRuntimeStatus vs. OrchestrationStatus in public API

### ISSUE THREAD

**Issue by f2bo** [Azure/azure-functions-durable-extension#1170]
We're developing an API for our app to monitor and manage the orchestration history. Since we don't want to expose durable function data types in our API, we represent the orchestration status using our own enum that's essentially a copy of `OrchestrationRuntimeStatus`.

I notice that there are two separate enums to represent the status of an orchestration, `DurableTask.Core.OrchestrationStatus` and `Microsoft.Azure.WebJobs.Extensions.DurableTask.OrchestrationRuntimeStatus`. They are almost identical so, presumably, `OrchestrationRuntimeStatus` avoids exposing DTF types directly from the Durable Functions API as well as including an additional `Unknown` state. In fact, `OrchestrationStatus` is not documented in the Durable Functions docs, reinforcing the idea that it was not meant to be exposed publicly.

https://docs.microsoft.com/en-us/dotnet/api/microsoft.azure.webjobs.extensions.durabletask.idurableorchestrationclient.purgeinstancehistoryasync

However, I'm a bit confused with the methods exposed by the `DurableOrchestrationClient`. On the one hand, `GetStatusAsync` return orchestration status using `OrchestrationRuntimeStatus`. On the other `PurgeInstanceHistoryAsync` uses `OrchestrationStatus`. This seems inconsistent and requires a mapping that appears unnecessary. 

Is this an oversight?

**Comment by cgillum** [comment]
Indeed. This appears to be a mistake. Thanks for bringing it to our attention.

**Comment by YunchuWang** [comment]
Thanks for reporting this. The inconsistency still exists in the legacy `IDurableOrchestrationClient` API, but correcting it would require a breaking change to an established public method signature.

The current .NET isolated `DurableTaskClient` APIs no longer have this mismatch: `PurgeInstancesFilter.Statuses` uses `OrchestrationRuntimeStatus`, consistently with the modern query and status APIs. Since support for the .NET in-process model ends on November 10, 2026, we don't plan to make a breaking API change in the legacy surface.

Closing as not planned. We recommend migrating to the .NET isolated worker APIs.
